### PR TITLE
feat: [CDS-77528]: support edit and delete actions for gitX webhooks

### PR DESCRIPTION
**Pull request by vikrant-harness**
### Summary

- support edit and delete actions for the gitx webhooks listing page in the action menu
- for edit only the identifier is non editable as discussed with PM 

#### Screenshots

https://github.com/harness/harness-core-ui/assets/106532291/c11cebfe-b46a-435f-95a9-afb8b7a53464

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Review by Sagar-Rajput7 — APPROVED** [review]

**Review by PratyushGar — APPROVED** [review]

File: src/modules/75-cd/pages/webhooks/NewWebhookModal.tsx [hunk 1]
```diff
@@ -34,30 +34,48 @@ export default function NewWebhookModal(props: NewWebhookModalProps): JSX.Elemen
...
-        webhook_identifier: values.identifier,
-        webhook_name: values.name
-      }
-    })
+    const folder_paths = values.folderPaths.map(path => path.value)
+
+    isEdit
+      ? updateWebhook({
+          'gitx-webhook': values.identifier,
+          body: {
```
Comment (pankajsingh-harness) [inline src/modules/75-cd/pages/webhooks/NewWebhookModal.tsx:49]: NIT: Common body param can be created as const and used at both places

File: src/modules/75-cd/pages/webhooks/NewWebhookModal.tsx [hunk 2]
```diff
@@ -34,30 +34,48 @@ export default function NewWebhookModal(props: NewWebhookModalProps): JSX.Elemen
...
-    createWebhook({
-      body: {
-        connector_ref: values.connectorRef,
-        folder_paths: folder_paths,
-        repo_name: values.repo,
-        webhook_identifier: values.identifier,
-        webhook_name: values.name
-      }
-    })
+    const folder_paths = values.folderPaths.map(path => path.value)
```
Comment (pankajsingh-harness) [inline src/modules/75-cd/pages/webhooks/NewWebhookModal.tsx:44]: Is this change expected? As per old code, folderPaths may be string also.

File: src/modules/75-cd/pages/webhooks/NewWebhookModal.tsx [hunk 3]
```diff
@@ -146,7 +161,11 @@ export default function NewWebhookModal(props: NewWebhookModalProps): JSX.Elemen
                 }}
               />
               <Layout.Horizontal spacing="small" padding={{ top: 'large' }}>
-                <Button variation={ButtonVariation.PRIMARY} type="submit" text={getString('add')} />
+                <Button
+                  variation={ButtonVariation.PRIMARY}
+                  type="submit"
+                  text={isEdit ? getString('save') : getString('add')}
```
Comment (pankajsingh-harness) [inline src/modules/75-cd/pages/webhooks/NewWebhookModal.tsx:167]: Do we not have RBAC concept for this flow?

File: src/modules/75-cd/pages/webhooks/WebhooksList/WebhooksList.tsx [hunk 4]
```diff
@@ -18,26 +26,92 @@ import {
...
+    })
+    setEditable(true)
+    setRowData(dataRow)
+    showCreateModal()
   }
-  const handleWebhookDelete = async (): Promise<void> => {
-    // TODO Delete Environment goes here
+  const handleWebhookDelete = async (name: string, identifier: string): Promise<void> => {
+    try {
+      await deleteGitxWebhook({
```
Comment (pankajsingh-harness) [inline src/modules/75-cd/pages/webhooks/WebhooksList/WebhooksList.tsx:102]: Is BE sending non 20* status codes for all error scenarios?

File: src/modules/75-cd/pages/webhooks/NewWebhookModal.tsx (hunk 2 above)
Comment (pankajsingh-harness) [reply src/modules/75-cd/pages/webhooks/NewWebhookModal.tsx:44]: This has been updated now.
```
folderPaths: { id: string; value: string }[]
```

File: src/modules/75-cd/pages/webhooks/NewWebhookModal.tsx (hunk 1 above)
Comment (vikrant-harness) [reply src/modules/75-cd/pages/webhooks/NewWebhookModal.tsx:49]: Will do this in next PR

File: src/modules/75-cd/pages/webhooks/NewWebhookModal.tsx (hunk 3 above)
Comment (vikrant-harness) [reply src/modules/75-cd/pages/webhooks/NewWebhookModal.tsx:167]: BE hasn't yet built the RBAC framework for the same , it's under TODO

File: src/modules/75-cd/pages/webhooks/WebhooksList/WebhooksList.tsx (hunk 4 above)
Comment (vikrant-harness) [reply src/modules/75-cd/pages/webhooks/WebhooksList/WebhooksList.tsx:102]: Which status codes @pankajsingh-harness ?

File: src/modules/75-cd/pages/webhooks/NewWebhookModal.tsx (hunk 2 above)
Comment (vikrant-harness) [reply src/modules/75-cd/pages/webhooks/NewWebhookModal.tsx:44]: yes this change is expected

**Review by pankajsingh-harness — APPROVED** [review]

**Comment by vikrant-harness** [comment]
Coverage will be added as a part of separate ticket

**Review by chetanvekariya-harness — APPROVED** [review]

**Comment by vikrant-harness** [comment]
https://harness.atlassian.net/browse/CDS-78807

**Pull request merged**